src/SqlPersistence/SynchronizedStorage/StorageAdapter.cs

```diff
@@ -43,11 +44,21 @@ public Task<CompletableSynchronizedStorageSession> TryAdapt(OutboxTransaction tr
             return session;
         }
 
-        Transaction existingTransaction;
+        Transaction transportTx;
         // Transport supports DTC and uses TxScope owned by the transport
-        if (transportTransaction.TryGet(out existingTransaction))
+        transportTransaction.TryGet(out transportTx);
+        if (transportTx != null && Transaction.Current != null && !transportTx.Equals(Transaction.Current))
+        {
+            throw new Exception("A TransctionScope has been opened in the current context overriding the one created by the transport. " 
```

Comment:
"If you wish to manually control the TransactionScope" > "To manually control the TransactionScope"